debugger/src/main/java/org/codice/acdebugger/api/SecuritySolution.java

```diff
@@ -88,44 +88,68 @@ protected SecuritySolution(SecuritySolution solution) {
     return Collections.unmodifiableList(doPrivileged);
   }
 
-  /** Prints information about this solution. */
-  public void print() {
-    print("");
+  /**
+   * Prints information about this solution.
+   *
+   * @param osgi <code>true</code> if attached to an OSGi container; <code>false</code> if not
+   */
+  public void print(boolean osgi) {
+    print(osgi, "");
   }
 
   /**
    * Prints information about this solution.
    *
+   * @param osgi <code>true</code> if attached to an OSGi container; <code>false</code> if not
    * @param prefix a prefix string to prepend to each printed line
    */
   @SuppressWarnings("squid:S106" /* this is a console application */)
-  public void print(String prefix) {
+  public void print(boolean osgi, String prefix) {
     System.out.println(ACDebugger.PREFIX + prefix + "{");
     final String and;
 
-    if (!grantedBundles.isEmpty()) {
-      final String s = (permissionInfos.size() == 1) ? "" : "s";
-
-      System.out.println(
-          ACDebugger.PREFIX
-              + prefix
-              + "    Add the following permission"
-              + s
-              + " to the appropriate policy file:");
-      System.out.println(
-          ACDebugger.PREFIX
-              + prefix
-              + "        grant codeBase \"file:/"
-              + grantedBundles.stream().sorted().collect(Collectors.joining("/"))
-              + "\" {");
-      permissionInfos
-          .stream()
-          .sorted()
-          .forEach(
-              p ->
-                  System.out.println(
-                      ACDebugger.PREFIX + prefix + "            permission " + p + ";"));
-      System.out.println(ACDebugger.PREFIX + prefix + "        }");
+    if (!grantedDomains.isEmpty()) {
+      if (osgi) {
```

Comment:
✏️ Probably worth breaking the osgi/non-osgi blocks out to their own private methods.